akka-persistence/src/main/scala/akka/persistence/journal/AsyncWriteJournal.scala

```diff
@@ -93,30 +94,29 @@ trait AsyncWriteJournal extends Actor with WriteJournalBase with AsyncRecovery {
       }
 
     case r @ ReplayMessages(fromSequenceNr, toSequenceNr, max, persistenceId, persistentActor) ⇒
-      // Send replayed messages and replay result to persistentActor directly. No need
-      // to resequence replayed messages relative to written and looped messages.
-      asyncReplayMessages(persistenceId, fromSequenceNr, toSequenceNr, max) { p ⇒
-        if (!p.deleted) // old records from 2.3 may still have the deleted flag
-          adaptFromJournal(p).foreach { adaptedPersistentRepr ⇒
-            persistentActor.tell(ReplayedMessage(adaptedPersistentRepr), Actor.noSender)
-          }
-      } map {
-        case _ ⇒ ReplayMessagesSuccess
-      } recover {
+
+      asyncReadHighestSequenceNr(persistenceId, fromSequenceNr).flatMap { highSeqNr ⇒
+        val toSeqNr = math.min(toSequenceNr, highSeqNr)
+        if (highSeqNr == 0L || fromSequenceNr > toSeqNr)
```

Comment:
Would a user randomly guess a number to start recovering from such that it would be wrong like (`>` triggers) this and still OK? My line of thinking was if we're recovery from a known sequenceId, and it's not there probably something fishy is going on.